### ISSUE TITLE
npm publish gives 500

### ISSUE THREAD

**Issue by tcurdt** [npm/npm#11465]
Given this is probably an OPS issue I am not sure if this is the right place but... I am trying to publish (a scoped package) and get a 500 error.

    npm publish  --access public

I am getting

    2 info using npm@3.5.3
    3 info using node@v5.4.1
    4 verbose publish [ '.' ]
    ...
    26 verbose publish registryBase https://registry.npmjs.org/
    27 silly publish uploading [...]/package.tgz
    28 verbose request uri https://registry.npmjs.org/@[...]
    29 verbose request sending authorization for write operation
    30 info attempt registry request try #1 at 11:16:56 PM
    31 verbose request using bearer token for auth
    32 verbose request id a2d7fcc2154f9e86
    33 http request PUT https://registry.npmjs.org/@[...]
    34 http 500 https://registry.npmjs.org/@[...]
    35 verbose headers { 'content-type': 'application/json',
    35 verbose headers   'cache-control': 'max-age=0',
    35 verbose headers   'content-length': '39',
    35 verbose headers   'accept-ranges': 'bytes',
    35 verbose headers   date: 'Sun, 07 Feb 2016 22:16:57 GMT',
    35 verbose headers   via: '1.1 varnish',
    35 verbose headers   connection: 'keep-alive',
    35 verbose headers   'x-served-by': 'cache-fra1225-FRA',
    35 verbose headers   'x-cache': 'MISS',
    35 verbose headers   'x-cache-hits': '0',
    35 verbose headers   'x-timer': 'S1454883416.743381,VS0,VE850' }
    36 info retry will retry, error on last attempt: Error: unexpected status code: 404 : @[...]
    ...

**Comment by ashleygwilliams** [comment]
hi! i've just tried to reproduce this with the same version of npm and node as you have, and i could not. a couple of qs:

- where are you connecting from?
- if you try again do you get the same error?

also, if you could share your `package.json` that'd be great. thanks (and sorry this is happening)!

**Comment by tcurdt** [comment]
I am connecting from Berlin, Germany. Oddly enough the traceroute does not really get me to the registry:

    $ traceroute registry.npmjs.org
    traceroute to a.sni.fastly.net (185.31.17.162), 64 hops max, 52 byte packets
     1  192.168.178.1 (192.168.178.1)  1623.010 ms  1.941 ms  1.272 ms
     2  87.186.224.23 (87.186.224.23)  18.701 ms  17.613 ms  18.438 ms
     3  217.0.75.230 (217.0.75.230)  20.952 ms  22.584 ms  21.243 ms
     4  b-ea7-i.b.de.net.dtag.de (62.154.47.82)  19.334 ms  21.562 ms
        b-ea7-i.b.de.net.dtag.de (62.154.46.186)  20.668 ms
     5  194.25.210.46 (194.25.210.46)  18.783 ms  63.576 ms  19.642 ms
     6  ae-1-60.edge5.frankfurt1.level3.net (4.69.154.9)  35.415 ms  35.949 ms
        ae-4-90.edge5.frankfurt1.level3.net (4.69.154.201)  43.269 ms
     7  ae-4-90.edge5.frankfurt1.level3.net (4.69.154.201)  42.996 ms
        ae-1-60.edge5.frankfurt1.level3.net (4.69.154.9)  36.357 ms  36.501 ms
     8  * * *

But I've also tried from from other locations (via VPN). Same behaviour for all retries and locations.

**Comment by tcurdt** [comment]
The package.json looks like this:

    {
      "version": "1.0.0",
      "author": "Torsten Curdt",
      "license": "Apache-2.0",
      "repository": {
        "type": "git",
        "url": "git://github.com/tcurdt/xstatic.git"
      },
      "engines": {
        "node": ">= 4.2.0"
      },
      "devDependencies": {
        "tape": "4.4.0",
        "blue-tape": "0.2.0",
        "istanbul": "0.4.2",
        "nyc": "5.5.0",
        "faucet": "0.0.1"
      },
      "name": "@xstatic/core",
      "dependencies": {
        "chokidar": "1.4.2",
        "minimatch": "3.0.0",
        "mkdirp": "0.5.1",
        "moment": "2.11.2",
        "browser-sync": "2.11.1",
        "@tcurdt/filepath": "1.0.0",
        "@tcurdt/tinyutils": "1.0.0"
      },
      "main": "lib/index.js",
      "files": [
        "lib/"
      ],
      "scripts": {
        "test": "istanbul test tape -- tests/*.js | faucet",
        "coverage": "nyc --reporter=text npm test | cat"
      }
    }

Could it be because of the scope `@xstatic`?

**Comment by tcurdt** [comment]
I don't "own" the scope yet I guess.

**Comment by tcurdt** [comment]
@ashleygwilliams anything else I could help with?

**Comment by tcurdt** [comment]
OK - seems like publishing to the @tcurdt scope works.
Not sure why @xstatic does not work.
Does a npm account have to exist for a scope to exist?

**Comment by zkat** [comment]
We really shouldn't be doing 500s like that for user-related/auth errors. :\

**Comment by chrisdickinson** [comment]
@tcurdt: Yes — a scope on the public registry must be backed by a user or organization. Were you originally publishing as the `xstatic` user, or another user?

@zkat: Agreed — I'll open a tracking ticket on registry-frontdoor.

**Comment by tcurdt** [comment]
@chrisdickinson thanks for clarifying! From the docs it wasn't clear that the scope has to match to a user/org. I tried to publish to the `xstatic` scope as `tcurdt`. The try to publish took very long and only then failed with a 500.

It would be great if the registry would fail immediately with a 4xx status code and preferably also give a clear message. On top of that it would be great to make the docs a bit clearer.

I later registered the xstatic user but for some reason could not login from the cli to push.
In the end I gave up on the scope idea as this really wasn't a smooth ride and already too much time wasted.

Anyway - thanks for the help. Feel free to close.
